src/pages/stats/[login].tsx

```diff
@@ -26,19 +28,34 @@ export default function Stats() {
   const [searchQuery, setSearchQuery] = useState<string>("");
 
   const filteredRepositories = useMemo(() => {
-    // putting the default value of the repositories
     if (!searchQuery) {
-      return repositories;
+      return hideOwnRepo
+        ? repositories?.filter(
+            // apply hideOwnRepo filter if checked
+            (repoData) =>
+              repoData.repository.owner.login !== session?.user.login
+          )
```

Comment:
The logic that filters out own repos is written here and replicated in lines 48-51.

Maybe we should either run it once on top of this function or extract the filtering logic in a separate function, what do you think?